aten/src/ATen/native/native_functions.yaml

```diff
@@ -5708,7 +5708,7 @@
 - func: roll(Tensor self, int[1] shifts, int[1] dims=[]) -> Tensor
   variants: function, method
   dispatch:
-    CPU: roll_cpu
+    CPU, MPS: roll
```

Comment:
This is not the way to fallback to CPU. You can register the Op in MPSFallback.mm to move it. The problem is that it will copy all the Tensors to CPU , then perform roll and move the resulting Tensor back to GPU timeline. This can be costly.

Reply:
We're not falling back to CPU, just reuse the same implementation, but on MPS device. (I think this can actually be changed to complicitexplicit if backends other than cpu and mps work fine)

Reply:
```
  auto t0 = self.narrow(dim, start, size-start);
  auto t1 = self.narrow(dim, 0, start);
  return at::cat({std::move(t0), std::move(t1)}, dim);
}
```
I see, it will use narrow and cat to implement it. @albanD , what do you think?

Reply:
We can't really use CompositeExplicit as CUDA has a separate one.
This sounds ok to me, just add a comment next to the function impl to mention that this is used both by CPU and MPS.

Reply:
Sorry for not being clear. I meant, except for CUDA, other backends go with CompositeExplicit. I do prefer to leave it as is since other backends would unlikely work out of the box I think.